roles/preflight-checks/tasks/check_items.yml

```diff
@@ -84,3 +84,15 @@
   when: v7k.enabled
   tags: ['precheck_v7k']
   run_once: true
+
+- name:  Make sure that if fixed_key is defined then cinder encryption is also enabled
+  assert:
+    that: "cinder.encryption.enabled"
+  when: cinder.encryption.fixed_key
+  run_once: true
+
+- name:  Make sure that if cinder encryption is enabled, then only barbican or fixed_key is used (XOR)
+  assert:
+    that: "barbican.enabled != cinder.encryption.fixed_key"
```

Comment:
`barbican.enabled != cinder.encryption.fixed_key` is always true, because barbican.enable is `true` or `false`, while cinder.encryption.fixed_key is a string or None. (cinder.encryption.fixed_key == None)